### ISSUE TITLE
Small question, do ALL canvas only support floats for XY(Z) positioning, widths, etc.

### ISSUE THREAD

**Issue by GhbSmwc** [godotengine/godot-docs#4480]
I check out the [canvas documentations](https://docs.godotengine.org/en/stable/classes/class_canvasitem.html) and it seems that only floating point are supported, despite [Vector2i](https://docs.godotengine.org/en/latest/classes/class_vector2i.html) and [Vector3i](https://docs.godotengine.org/en/latest/classes/class_vector3i.html) exists.

**Comment by Calinou** [comment]
Vector2i and Vector3i are only available in `master` (what will become Godot 4.0), not 3.2.x.

Also, yes, floating-point canvas coordinates are allowed.

**Comment by GhbSmwc** [comment]
@Calinou How does godot render between-pixels? Unless it simply rounds downwards. Good thing these precision loss will only occur far offscreen (no monitors would have 2^52 or 2^23 number of pixels wide and tall)

**Comment by Calinou** [comment]
> How does godot render between-pixels? Unless it simply rounds downwards

By default, it doesn't attempt to round pixel coordinates. This means things can move smoothly on screen, but it'll also make things blurrier. You can enable pixel snap to make sure all pixel transforms are snapped before rendering.

**Comment by GhbSmwc** [comment]
@Calinou So Godot actually supports displaying “sub-pixels”. Cool.

**Comment by GhbSmwc** [comment]
Also, according to [this](https://gist.github.com/Calinou/49aefe52ce8f67ffa3f743932123d14f), will many other things related to textures being drawn, such as [controls](https://docs.godotengine.org/en/stable/classes/class_control.html)'s properties (margin, size, position, global position etc.)?

**Comment by Calinou** [comment]
> Also, according to this, will many other things related to textures being drawn, such as controls's properties (margin, size, position, global position etc.)?

No, as Controls are snapped to pixels by default, even if the properties themselves allow floating-point values. You can disable this by unchecking Snap Controls To Pixels in the Project Settings.

**Comment by GhbSmwc** [comment]
@Calinou Ack, that last sentence was a fragment, I was meant to say:

“Also, according to this, will many other things related to textures being drawn, such as controls's properties (margin, size, position, global position etc.) support integer?”

One last question: When pixel snapping is turned on, does internally that they're handled as ints (if you set it to a fractional value, it rounds into a whole number upon type conversion)? This is in case if you are reading the position of a rectangle after setting it to X=0.5 and it returns 0.

**Comment by Calinou** [comment]
> One last question: When pixel snapping is turned on, does internally that they're handled as ints (if you set it to a fractional value, it rounds into a whole number upon type conversion)? This is in case if you are reading the position of a rectangle after setting it to X=0.5 and it returns 0.

I don't remember, but the way transforms are snapped was revised for 4.0 in https://github.com/godotengine/godot/pull/43554.
